src/ServiceControl/ExternalIntegrations/MessageFailedPublisher.cs

```diff
@@ -27,10 +27,9 @@ protected override async Task<IEnumerable<object>> PublishEvents(IEnumerable<Dis
             var failedMessages = new List<object>(failedMessageData.Length);
             foreach (var entity in failedMessageData)
             {
-                session.Advanced.Evict(entity);
-
                 if (entity != null)
```

Comment:
Another option would be to add a `where` clause to the list and then iterate over that but I prefer the `if` approach

Reply:
I had that originally, but then saw the if that was in the loop already and thought it looked neater than including the where in the loop above.